### PR TITLE
Improve Code Coverage on EnumConverter

### PR DESCRIPTION
**Pull request by Jacksondr5**
- Helps address #32341 
- Adds coverage to branch where WriteWithQuotes goes over the name cache soft limit. [code](https://github.com/dotnet/runtime/blob/e0913bd003eaa3fef6ef11c5d1380ca5005b1253/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs#L356-L361) 
- Adds coverage to branches where a Dictionary key is a numeric enum that is not in the cache and the value is not a valid identifier.  [code](https://github.com/dotnet/runtime/blob/e0913bd003eaa3fef6ef11c5d1380ca5005b1253/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs#L366-L395)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39561) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39561) before we can accept your contribution.<br/><br/>:x: Jacksondr5 [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39561)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=39561) it.</sub>

**Comment by Jacksondr5** [comment]
I looked at writing a test for [the constructer soft check](https://github.com/dotnet/runtime/blob/e0913bd003eaa3fef6ef11c5d1380ca5005b1253/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs#L56-L59), but I'm not sure if it's desired.  Given that the other tests that cover the cache limit serialize up to 2097152 enums, I think we would need to have a hilariously large enum literal to truly test that the constructor code doesn't throw an OutOfMemoryException.  Maybe I could do something with reflection, but I wanted to check before I wrote anything.

Looks like there's a bug in how the coverage reporter interprets switch statements
image
Hovering over the branch icon on L245 shows that 9 of 11 branches are covered.  Given that it's 7 PM on a Friday for me, I don't want to make any strong claims involving math, but those numbers don't seem right to me.

Are we able to cover lines that come after ThrowHelper statements?  They look unreachable to me.

**Comment by Jacksondr5** [comment]
I believe the checks were failing due to a build problem last week that looks to be resolved now.  I don't think I have permissions to re-run the builds, can someone trigger that for me?

File: src/libraries/System.Text.Json/tests/Serialization/EnumConverterTests.cs [hunk 1]
```diff
@@ -294,5 +295,230 @@ public enum MyEnum
...
+        public enum SampleEnumInt32
+        {
+            A = 1 << 0,
+            B = 1 << 1,
+            C = 1 << 2,
+            D = 1 << 3,
+            E = 1 << 4,
+            F = 1 << 5,
+            G = 1 << 6,
+            H = 1 << 7,
```
Comment (Jozkee) [inline src/libraries/System.Text.Json/tests/Serialization/EnumConverterTests.cs:430]: Seems like `H` is never used by `SerilizeDictionaryWhenCacheIsFull`, you should remove it here and in the other enums.

**Comment by Jozkee** [comment]
> Hovering over the branch icon on L245 shows that 9 of 11 branches are covered. Given that it's 7 PM on a Friday for me, I don't want to make any strong claims involving math, but those numbers don't seem right to me.

I agree, the switch expression only contains 9 cases, I don't see how the coverage reports 9/11 branches, I would expect 8/9 since I don't think we can even cover the default case. Consider open an issue in https://github.com/danielpalme/ReportGenerator.

> Are we able to cover lines that come after ThrowHelper statements? They look unreachable to me.

That code is unreachable because we throw but we still need to return in all code paths.

**Comment by Jacksondr5** [comment]
Dug a bit deeper, the coverage branching issue actually seems to be a problem with [coverlet](https://github.com/coverlet-coverage/coverlet/issues/903).

**Comment by Jacksondr5** [comment]
Think I messed that one up..... I'll open another

**Pull request closed without merging**